src/N98/Magento/Command/Cache/ReportCommand.php

```diff
@@ -0,0 +1,162 @@
+{
+    /**
+     * @var CacheInterface
+     */
+    private $cache;
+
+    /**
+     * @var FullPageCache
+     */
+    private $fpc;
+
+    /**
+     * @param array $metadata
+     * @param InputInterface $input
+     * @return bool
+     */
+    private function isTagFiltered($metadata, $input)
+    {
+        return (bool) count(
+            array_intersect(
+                $metadata['tags'],
+                explode(',', $input->getOption('filter-tag'))
+            )
+        );
+    }
+
+    /**
+     * @param CacheInterface $cache
+     * @param FullPageCache $fpc
+     */
+    public function inject(
+        CacheInterface $cache,
+        FullPageCache $fpc
+    ) {
+        $this->cache = $cache;
+        $this->fpc = $fpc;
+    }
+
+    /**
+     * @return void
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('cache:report')
+            ->addOption(
+                'fpc',
+                null,
+                InputOption::VALUE_NONE,
+                'Use full page cache instead of core cache'
+            )
+            ->addOption(
+                'tags',
+                't',
+                InputOption::VALUE_NONE,
+                'Output tags'
+            )
+            ->addOption(
+                'mtime',
+                'm',
+                InputOption::VALUE_NONE,
+                'Output last modification time'
+            )
+            ->addOption(
+                'filter-id',
+                '',
+                InputOption::VALUE_OPTIONAL,
+                'Filter output by ID (substring)'
+            )
+            ->addOption(
+                'filter-tag',
+                '',
+                InputOption::VALUE_OPTIONAL,
+                'Filter output by TAG (separate multiple tags by comma)'
+            )
+            ->addOption(
+                'format',
+                null,
+                InputOption::VALUE_OPTIONAL,
+                'Output Format. One of [' . implode(', ', RendererFactory::getFormats()) . ']'
+            )
+            ->setDescription('View inside the cache');
+    }
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @return int|void
+     * @throws RuntimeException
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->detectMagento($output, true);
+        if (!$this->initMagento()) {
+            return;
+        }
+
+        /** @var \Zend_Cache_Core $lowLevelFrontend */
+        if ($input->hasOption('fpc') && $input->getOption('fpc')) {
+            $lowLevelFrontend = $this->fpc->getLowLevelFrontend();
+        } else {
+            $lowLevelFrontend = $this->cache->getFrontend()->getLowLevelFrontend();
+        }
+
+        $table = [];
+        $cacheIds = $lowLevelFrontend->getIds();
+        foreach ($cacheIds as $cacheId) {
+            if ($input->getOption('filter-id') !== null &&
```

Comment:
Depending on how large such a cache-IDs list could be, repeatedly calling `$input->getOption()` for the same values might start to hurt. I can't say, you probably know better. If not let's keep it until actual problems arise.